meta.go

```diff
@@ -273,8 +276,46 @@ type Meta struct {
 	Resources      map[string]resource.Meta `bson:"resources,omitempty" json:"Resources,omitempty"`
 	Terms          []string                 `bson:"terms,omitempty" json:"Terms,omitempty"`
 	MinJujuVersion version.Number           `bson:"min-juju-version,omitempty" json:"min-juju-version,omitempty"`
+
+	Systems       []systems.System     `bson:"systems,omitempty" json:"systems,omitempty" yaml:"systems,omitempty"`
+	Platforms     []Platform           `bson:"platforms,omitempty" json:"platforms,omitempty" yaml:"platforms,omitempty"`
+	Architectures []Architecture       `bson:"architectures,omitempty" json:"architectures,omitempty" yaml:"architectures,omitempty"`
+	Containers    map[string]Container `bson:"containers,omitempty" json:"containers,omitempty" yaml:"containers,omitempty"`
 }
 
+type Platform string
+
+const (
+	PlatformMachine    Platform = "machine"
+	PlatformKubernetes Platform = "kubernetes"
+)
+
+type Architecture string
+
+const (
+	AMD64   Architecture = "amd64"
+	ARM64   Architecture = "arm64"
+	PPC64EL Architecture = "ppc64el"
+	S390X   Architecture = "s390x"
+)
```

Comment:
We have arch stuff under juju/utils.
Ideally one day we'd move it to juju/arch and get this from there

Reply:
Lets come back to this when architecture support is added to the rest of juju